### PR TITLE
Upgrade to rails 4.2

### PR DESCRIPTION
**Pull request by gsrohde**
This addresses security vulnerabilities in Rails 4.1.

@robkooper You can switch the test instance to this branch.

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -33,20 +35,22 @@
...
+  # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for NGINX
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
 
-  # Set to :debug to see everything in the log.
-  config.log_level = :info
+  # Use the lowest log level to ensure availability of diagnostic information
+  # when problems arise.
+  config.log_level = :debug
```
Comment (robkooper) [inline config/environments/production.rb]: Should this really be debug?

**Comment by robkooper** [comment]
Running on http://pecan-dev.ncsa.illinois.edu/bety/

**Comment by dlebauer** [comment]
Should http://pecan-dev.ncsa.illinois.edu/bety/trait_covariate_associations be empty?

**Comment by gsrohde** [comment]
@robkooper 

From the latest Rails Guide on configuration: "config.log_level defines the verbosity of the Rails logger. This option defaults to :debug for all environments. The available log levels are: :debug, :info, :warn, :error, :fatal, and :unknown."

From the Rails 3.2 Guide: "config.log_level defines the verbosity of the Rails logger. This option defaults to :debug for all modes except production, where it defaults to :info."

I'm not sure why the default changed—it puzzled me too when I noticed it.  I guess I'll set it to :info for production.

**Pull request merged**